src/radical/pilot/utils/session.py

```diff
@@ -93,6 +94,49 @@ def fetch_profiles (sid, dburl=None, client=None, tgt=None, access=None, session
             print "Overriding remote sandbox: %s" % sandbox_url
 
         sandbox  = saga.filesystem.Directory (sandbox_url, session=session)
+
+        # Try to fetch a tarball of profiles, so that we can get them all in one (SAGA) go!
+        PROFILES_TARBALL = 'profiles.tgz'
+        tarball_available = False
+        try:
+            if sandbox.is_file(PROFILES_TARBALL):
+                print "Profiles tarball exists!"
+
+                ftgt = saga.Url('%s/%s' % (tgt_url, PROFILES_TARBALL))
+
+                if skip_existing and os.path.isfile(ftgt.path) \
+                        and os.stat(ftgt.path).st_size > 0:
+
+                    print "Skipping fetching of '%s/%s' to '%s'." % (sandbox_url, PROFILES_TARBALL, tgt_url)
+                    tarball_available = True
+                else:
+
+                    print "Fetching '%s%s' to '%s'." % (sandbox_url, PROFILES_TARBALL, tgt_url)
+                    prof_file = saga.filesystem.File("%s%s" % (sandbox_url, PROFILES_TARBALL), session=session)
+                    prof_file.copy(ftgt, flags=saga.filesystem.CREATE_PARENTS)
+                    prof_file.close()
+
+                    tarball_available = True
+            else:
+                print "Profiles tarball doesnt exists!"
+
+        except saga.DoesNotExist:
+            print "exception(TODO): profiles tarball doesnt exists!"
+
+        # We now have a local tarball
+        if tarball_available:
+            print "Extracting tarball %s into '%s'." % (ftgt.path, tgt_url.path)
+            tarball = tarfile.open(ftgt.path)
+            tarball.extractall(tgt_url.path)
+
+            profiles = glob.glob("%s/*.prof" % tgt_url.path)
+            print "Tarball %s extracted to '%s'." % (ftgt.path, profiles)
+            ret.extend(profiles)
+
+            # If extract succeeded, no need to fetch individual profiles
+            continue
+
+        # If we dont have a tarball (for whichever reason), fetch individual profiles
```

Comment:
What is the default delay between the two?  Would it be (reasonably) safe to assume that tar (and other shutdown activities such as profile and log flushing, mongodb state updates) are not interrupted?